Description: Fix build on Debian/kFreeBSD
From: Nicolas Braud-Santoni <nicolas@braud-santoni.eu>
Reviewed-by: Alessio Di Mauro <alessio@yubico.com>
Forwarded: https://github.com/Yubico/pam-u2f/pull/40
Applied-Upstream: yes
Last-Update: 2016-08-02

From e834c2cbbbd08379b48a798d46b4089029f127b9 Mon Sep 17 00:00:00 2001
From: Nicolas Braud-Santoni <nicolas@braud-santoni.eu>
Date: Mon, 1 Aug 2016 22:41:07 +0200
Subject: [PATCH] pam-u2f.c: Define _GNU_SOURCE to make secure_getenv(3)
 available

Closes #37

Credit goes to Steven Chamberlain and Helmut Grohne
for finding the root cause.
---
 pam-u2f.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pam-u2f.c b/pam-u2f.c
index 8fbb647..e94202f 100644
--- a/pam-u2f.c
+++ b/pam-u2f.c
@@ -7,7 +7,7 @@
  *__USE_GNU defined and not get a warning about
  * secure_getenv during compilation
  */
-#ifdef __linux__
+#ifdef HAVE_SECURE_GETENV
 #define _GNU_SOURCE
 #endif
 
-- 
2.8.1

